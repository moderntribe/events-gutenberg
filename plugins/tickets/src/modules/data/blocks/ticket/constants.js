export const TC = 'tribe_commerce';
export const EDD = 'easy_digital_downloads';
export const WOO = 'wooCommerce';

export const TC_CLASS = 'Tribe__Tickets__Commerce__PayPal__Main';
export const EDD_CLASS = 'Tribe__Tickets_Plus__Commerce__EDD__Main';
export const WOO_CLASS = 'Tribe__Tickets_Plus__Commerce__WooCommerce__Main';

export const PROVIDER_CLASS_TO_CONSTANT_MAPPING = {
	[ TC_CLASS ]: TC,
	[ EDD_CLASS ]: EDD,
	[ WOO_CLASS ]: WOO,
};

export const PROVIDER_TYPES = [ TC, EDD, WOO ];
