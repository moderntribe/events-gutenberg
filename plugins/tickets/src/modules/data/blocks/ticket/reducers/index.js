/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import ui from './ui';
import tickets from './tickets';

export default combineReducers( {
	ui,
	tickets,
} );
