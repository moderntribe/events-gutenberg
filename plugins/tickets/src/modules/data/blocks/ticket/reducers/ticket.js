/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import * as types from './../types';
import { TICKET_TYPES } from '@moderntribe/tickets/data/utils';
import { moment as momentUtil } from '@moderntribe/common/utils';

const currentMoment = moment();
const ADDITIONAL_DAYS = 3;
export const DEFAULT_STATE = {
	title: '',
	description: '',
	price: '',
	SKU: '',
	startDate: momentUtil.toDate( currentMoment ),
	startDateMoment: currentMoment,
	endDate: momentUtil.toDate( currentMoment.clone().add( ADDITIONAL_DAYS, 'days' ) ),
	endDateMoment: currentMoment,
	startTime: momentUtil.toTime24Hr( currentMoment ),
	endTime: momentUtil.toTime24Hr( currentMoment ),
	dateIsPristine: true,
	capacityType: TICKET_TYPES.independent,
	capacity: '',
	isEditing: false,
	ticketId: 0,
	sold: 0,
	isLoading: false,
	hasBeenCreated: false,
	available: 0,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_TICKET_DESCRIPTION:
			return {
				...state,
				description: action.payload.description,
			};
		case types.SET_TICKET_PRICE:
			return {
				...state,
				price: action.payload.price,
			};
		case types.SET_TICKET_SKU:
			return {
				...state,
				SKU: action.payload.SKU,
			};
		case types.SET_TICKET_START_DATE:
			return {
				...state,
				startDate: action.payload.startDate,
			};
		case types.SET_TICKET_START_TIME:
			return {
				...state,
				startTime: action.payload.startTime,
			};
		case types.SET_TICKET_END_DATE:
			return {
				...state,
				endDate: action.payload.endDate,
			};
		case types.SET_TICKET_END_TIME:
			return {
				...state,
				endTime: action.payload.endTime,
			};
		case types.SET_TICKET_CAPACITY:
			return {
				...state,
				capacity: action.payload.capacity,
			};
		case types.SET_TICKET_CAPACITY_TYPE:
			return {
				...state,
				capacityType: action.payload.capacityType,
			};
		case types.SET_TICKET_IS_EDITING:
			return {
				...state,
				isEditing: action.payload.isEditing,
			};
		case types.SET_TICKET_ID:
			return {
				...state,
				ticketId: action.payload.ticketId,
			};
		case types.SET_TICKET_DATE_PRISTINE:
			return {
				...state,
				dateIsPristine: action.payload.dateIsPristine,
			};
		case types.SET_TICKET_START_DATE_MOMENT:
			return {
				...state,
				startDateMoment: action.payload.startDateMoment,
			};
		case types.SET_TICKET_END_DATE_MOMENT:
			return {
				...state,
				endDateMoment: action.payload.endDateMoment,
			};
		case types.SET_TICKET_IS_LOADING:
			return {
				...state,
				isLoading: action.payload.isLoading,
			};
		case types.SET_TICKET_HAS_BEEN_CREATED:
			return {
				...state,
				hasBeenCreated: action.payload.hasBeenCreated,
			};
		case types.SET_TICKET_SOLD:
			return {
				...state,
				sold: action.payload.sold,
			};
		case types.SET_TICKET_AVAILABLE:
			return {
				...state,
				available: action.payload.available,
			};
		default:
			return state;
	}
};
