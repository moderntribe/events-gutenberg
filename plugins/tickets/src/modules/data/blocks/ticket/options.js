/**
 * External dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { SHARED, INDEPENDENT, UNLIMITED, TICKET_TYPES } from './constants'

export const CAPACITY_TYPE_OPTIONS = [
	{
		label: __( 'Share capacity with other tickets', 'events-gutenberg' ),
		value: TICKET_TYPES[ SHARED ],
	}, {
		label: __( 'Set capacity for this ticket only', 'events-gutenberg' ),
		value: TICKET_TYPES[ INDEPENDENT ],
	}, {
		label: __( 'Unlimited', 'events-gutenberg' ),
		value: TICKET_TYPES[ UNLIMITED ],
	},
];
