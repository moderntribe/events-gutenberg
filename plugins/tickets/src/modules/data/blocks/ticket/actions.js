/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const setHeader = ( header ) => ( {
	type: types.SET_TICKET_HEADER,
	payload: {
		header,
	},
} );

export const setTotalSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TOTAL_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setSettingsOpen = ( isSettingsOpen ) => ( {
	type: types.SET_TICKET_SETTINGS_OPEN,
	payload: {
		isSettingsOpen,
	},
} );

export const openSettings = () => setSettingsOpen( true );
export const closeSettings = () => setSettingsOpen( false );

export const setParentBlockSelected = ( selected ) => ( {
	type: types.SET_PARENT_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setChildBlockSelected = ( selected ) => ( {
	type: types.SET_CHILD_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setActiveChildBlockId = ( activeChildBlockId ) => ( {
	type: types.SET_ACTIVE_CHILD_BLOCK_ID,
	payload: {
		activeChildBlockId,
	},
} );

export const setTempSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TMP_TICKET_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

export const setProvider = ( provider ) => ( {
	type: types.SET_PROVIDER,
	payload: {
		provider,
	},
} );

// individual ticket actions
export const registerTicketBlock = ( blockId ) => ( {
	type: types.SET_TICKET_BLOCK_ID,
	payload: {
		blockId,
	},
} );

export const removeTicketBlock = ( blockId ) => ( {
	type: types.REMOVE_TICKET_BLOCK,
	payload: {
		blockId,
	},
} );

export const setTitle = ( blockId, title ) => ( {
	type: types.SET_TICKET_TITLE,
	payload: {
		blockId,
		title,
	},
} );

export const setDescription = ( blockId, description ) => ( {
	type: types.SET_TICKET_DESCRIPTION,
	payload: {
		blockId,
		description,
	},
} );

export const setPrice = ( blockId, price ) => ( {
	type: types.SET_TICKET_PRICE,
	payload: {
		blockId,
		price,
	},
} );

export const setSKU = ( blockId, SKU ) => ( {
	type: types.SET_TICKET_SKU,
	payload: {
		blockId,
		SKU,
	},
} );

export const setStartDate = ( blockId, startDate ) => ( {
	type: types.SET_TICKET_START_DATE,
	payload: {
		blockId,
		startDate,
	},
} );

export const setStartTime = ( blockId, startTime ) => ( {
	type: types.SET_TICKET_START_TIME,
	payload: {
		blockId,
		startTime,
	},
} );

export const setEndDate = ( blockId, endDate ) => ( {
	type: types.SET_TICKET_END_DATE,
	payload: {
		blockId,
		endDate,
	},
} );

export const setEndTime = ( blockId, endTime ) => ( {
	type: types.SET_TICKET_END_TIME,
	payload: {
		blockId,
		endTime,
	},
} );

export const setCapacityType = ( blockId, capacityType ) => ( {
	type: types.SET_TICKET_CAPACITY_TYPE,
	payload: {
		blockId,
		capacityType,
	},
} );

export const setCapacity = ( blockId, capacity ) => ( {
	type: types.SET_TICKET_CAPACITY,
	payload: {
		blockId,
		capacity,
	},
} );

export const createNewTicket = ( blockId ) => ( {
	type: types.SET_CREATE_NEW_TICKET,
	payload: {
		blockId,
	},
} );

export const updateTicket = ( blockId ) => ( {
	type: types.SET_UPDATE_TICKET,
	payload: {
		blockId,
	},
} );

export const setTicketIsEditing = ( blockId, isEditing ) => ( {
	type: types.SET_TICKET_IS_EDITING,
	payload: {
		blockId,
		isEditing,
	},
} );

export const setTicketId = ( blockId, postId ) => ( {
	type: types.SET_TICKET_ID,
	payload: {
		blockId,
		postId,
	},
} );

export const setTicketDateIsPristine = ( blockId, dateIsPristine ) => ( {
	type: types.SET_TICKET_DATE_PRISTINE,
	payload: {
		blockId,
		dateIsPristine,
	},
} );

export const setParentBlockIsLoading = ( isParentBlockLoading ) => ( {
	type: types.SET_PARENT_BLOCK_LOADING,
	payload: {
		isParentBlockLoading,
	},
} );

export const setTicketStartDateMoment = ( blockId, startDateMoment ) => ( {
	type: types.SET_TICKET_START_DATE_MOMENT,
	payload: {
		blockId,
		startDateMoment,
	},
} );

export const setTicketEndDateMoment = ( blockId, endDateMoment ) => ( {
	type: types.SET_TICKET_END_DATE_MOMENT,
	payload: {
		blockId,
		endDateMoment,
	},
} );
