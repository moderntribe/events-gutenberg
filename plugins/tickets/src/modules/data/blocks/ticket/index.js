/**
 * Internal dependencies
 */
import reducer from './reducers';

import * as constants from './constants';
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import sagas from './sagas';

export default reducer;

export { constants, types, actions, selectors, sagas };
