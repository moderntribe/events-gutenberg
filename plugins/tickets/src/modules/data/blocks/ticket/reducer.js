/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const DEFAULT_STATE = {
	sharedCapacity: '',
	header: null,
	isSettingsOpen: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_HEADER:
			return {
				...state,
				header: action.payload.header,
			};
		case types.SET_TICKET_SHARED_CAPACITY:
			return {
				...state,
				sharedCapacity: action.payload.sharedCapacity,
			};
		case types.SET_TICKET_SETTINGS_OPEN:
			return {
				...state,
				isSettingsOpen: action.payload.isSettingsOpen,
			};
		default:
			return state;
	}
}
