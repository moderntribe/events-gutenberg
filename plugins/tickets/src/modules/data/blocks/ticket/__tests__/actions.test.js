/**
 * Internal dependencies
 */
import { actions } from '@moderntribe/tickets/data/blocks/ticket';

describe( 'Gutenberg actions', () => {
	describe( 'Initial state', () => {
		const props = {
			clientId: 'modern-tribe',
			attributes: {
				header: 99,
				sharedCapacity: 19,
			},
		};
		expect( actions.setInitialState( props ) ).toMatchSnapshot();
	} );

	describe( 'Parent block is loading', () => {
		expect( actions.setParentBlockIsLoading( true ) ).toMatchSnapshot();
		expect( actions.setParentBlockIsLoading( false ) ).toMatchSnapshot();
	} );

	describe( 'Header actions', () => {
		test( 'Set null to remove header', () => {
			expect( actions.setHeader( null ) ).toMatchSnapshot();
		} );

		test( 'Set object into header', () => {
			expect( actions.setHeader( { image: 10 } ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Set shared capacity actions', () => {
		test( 'Set empty value', () => {
			expect( actions.setTotalSharedCapacity( 0 ) ).toMatchSnapshot();
		} );

		test( 'Set a large number', () => {
			expect( actions.setTotalSharedCapacity( 999 ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Set the status of the settings dashboard', () => {
		test( 'Open the settings dashboard', () => {
			expect( actions.openSettings() ).toMatchSnapshot();
			expect( actions.setSettingsOpen( true ) ).toMatchSnapshot();
		} );

		test( 'Close the settings dashboard', () => {
			expect( actions.closeSettings() ).toMatchSnapshot();
			expect( actions.setSettingsOpen( false ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Block selection', () => {
		test( 'Parent block selection', () => {
			expect( actions.setParentBlockSelected( true ) ).toMatchSnapshot();
			expect( actions.setParentBlockSelected( false ) ).toMatchSnapshot();
		} );

		test( 'Child block selection', () => {
			expect( actions.setChildBlockSelected( true ) ).toMatchSnapshot();
			expect( actions.setChildBlockSelected( false ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Active block selected', () => {
		test( 'Active block', () => {
			expect( actions.setActiveChildBlockId( 'modern-tribe' ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Temporarily values', () => {
		test( 'Temporarily capacity', () => {
			expect( actions.setTempSharedCapacity( 99 ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Ticket Provider', () => {
		test( 'Set default provider', () => {
			expect( actions.setProvider( 'Tribe__Tickets__Commerce__PayPal__Main' ) ).toMatchSnapshot();
		} );
	} );

	describe( 'Single block actions', () => {
		const blockId = 'modern-tribe';

		test( 'Block registration process', () => {
			expect( actions.registerTicketBlock( blockId ) ).toMatchSnapshot();
			expect( actions.removeTicketBlock( blockId ) ).toMatchSnapshot();
		} );

		test( 'Ticket title', () => {
			expect( actions.setTitle( blockId, 'Modern Tribe' ) );
		} );

		test( 'Ticket description', () => {
			expect( actions.setDescription( blockId, 'The Next Generation of Digital Agency' ) )
				.toMatchSnapshot();
		} );

		test( 'Ticket price', () => {
			expect( actions.setPrice( blockId, 99 ) ).toMatchSnapshot();
		} );

		test( 'Ticket SKU', () => {
			expect( actions.setSKU( blockId, 'my-sku' ) ).toMatchSnapshot();
		} );

		test( 'Ticket dates', () => {
			expect( actions.setStartDate( blockId, 'January 1, 2018' ) ).toMatchSnapshot();
			expect( actions.setStartTime( blockId, '10:00' ) ).toMatchSnapshot();
			expect( actions.setEndDate( blockId, 'January 10, 2018' ) ).toMatchSnapshot();
			expect( actions.setEndTime( blockId, '12:34' ) ).toMatchSnapshot();
		} );

		test( 'Ticket Capacity', () => {
			expect( actions.setCapacity( blockId, 'unlimited' ) ).toMatchSnapshot();
			expect( actions.setCapacityType( blockId, 'unlimited' ) ).toMatchSnapshot();
		} );

		test( 'Create a new ticket', () => {
			expect( actions.createNewTicket( blockId ) ).toMatchSnapshot();
		} );

		test( 'Update ticket', () => {
			expect( actions.updateTicket( blockId ) ).toMatchSnapshot();
		} );

		test( 'Set ticket date is pristine', () => {
			expect( actions.setTicketDateIsPristine( blockId, false ) ).toMatchSnapshot();
			expect( actions.setTicketDateIsPristine( blockId, true ) ).toMatchSnapshot();
		} );

		test( 'Set ticket post ID', () => {
			expect( actions.setTicketId( blockId, 99 ) ).toMatchSnapshot();
		} );

		test( 'Ticket editing flag', () => {
			expect( actions.setTicketIsEditing( true ) ).toMatchSnapshot();
			expect( actions.setTicketIsEditing( false ) ).toMatchSnapshot();
		} );

		test( 'Ticket start moment object', () => {
			expect( actions.setTicketStartDateMoment( blockId, { type: 'moment' } ) ).toMatchSnapshot();
			expect( actions.setTicketStartDateMoment( blockId, null ) ).toMatchSnapshot();
		} );

		test( 'Ticket end moment object', () => {
			expect( actions.setTicketEndDateMoment( blockId, { type: 'moment' } ) ).toMatchSnapshot();
			expect( actions.setTicketEndDateMoment( blockId, null ) ).toMatchSnapshot();
		} );

		test( 'Ticket has been created', () => {
			expect( actions.seTicketHasBeenCreated( blockId, true ) ).toMatchSnapshot();
			expect( actions.seTicketHasBeenCreated( blockId, false ) ).toMatchSnapshot();
		} );

		test( 'Ticket is loading', () => {
			expect( actions.setTicketIsLoading( blockId, true ) ).toMatchSnapshot();
			expect( actions.setTicketIsLoading( blockId, false ) ).toMatchSnapshot();
		} );

		test( 'fetch ticket details', () => {
			expect( actions.fetchTicketDetails( blockId, 99 ) ).toMatchSnapshot();
		} );

		test( 'cancel ticket edit', () => {
			expect( actions.cancelTicketEdit( blockId ) ).toMatchSnapshot();
		} );
	} );
} );
