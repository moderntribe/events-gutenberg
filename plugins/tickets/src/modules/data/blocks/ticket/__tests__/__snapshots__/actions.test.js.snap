// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
Object {
  "payload": Object {
    "attributes": Object {
      "header": 99,
      "sharedCapacity": 19,
    },
    "clientId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_INITIAL_STATE",
}
`;

exports[` 2`] = `
Object {
  "payload": Object {
    "isParentBlockLoading": true,
  },
  "type": "@@MT/TICKETS/SET_PARENT_BLOCK_LOADING",
}
`;

exports[` 3`] = `
Object {
  "payload": Object {
    "isParentBlockLoading": false,
  },
  "type": "@@MT/TICKETS/SET_PARENT_BLOCK_LOADING",
}
`;

exports[`Gutenberg actions Active block selected Active block 1`] = `
Object {
  "payload": Object {
    "activeChildBlockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_ACTIVE_CHILD_BLOCK_ID",
}
`;

exports[`Gutenberg actions Block selection Child block selection 1`] = `
Object {
  "payload": Object {
    "selected": true,
  },
  "type": "@@MT/TICKETS/SET_CHILD_BLOCK_SELECTED",
}
`;

exports[`Gutenberg actions Block selection Child block selection 2`] = `
Object {
  "payload": Object {
    "selected": false,
  },
  "type": "@@MT/TICKETS/SET_CHILD_BLOCK_SELECTED",
}
`;

exports[`Gutenberg actions Block selection Parent block selection 1`] = `
Object {
  "payload": Object {
    "selected": true,
  },
  "type": "@@MT/TICKETS/SET_PARENT_BLOCK_SELECTED",
}
`;

exports[`Gutenberg actions Block selection Parent block selection 2`] = `
Object {
  "payload": Object {
    "selected": false,
  },
  "type": "@@MT/TICKETS/SET_PARENT_BLOCK_SELECTED",
}
`;

exports[`Gutenberg actions Header actions Set null to remove header 1`] = `
Object {
  "payload": Object {
    "header": null,
  },
  "type": "@@MT/TICKETS/SET_TICKET_HEADER",
}
`;

exports[`Gutenberg actions Header actions Set object into header 1`] = `
Object {
  "payload": Object {
    "header": Object {
      "image": 10,
    },
  },
  "type": "@@MT/TICKETS/SET_TICKET_HEADER",
}
`;

exports[`Gutenberg actions Set shared capacity actions Set a large number 1`] = `
Object {
  "payload": Object {
    "sharedCapacity": 999,
  },
  "type": "@@MT/TICKETS/SET_TICKET_TOTAL_SHARED_CAPACITY",
}
`;

exports[`Gutenberg actions Set shared capacity actions Set empty value 1`] = `
Object {
  "payload": Object {
    "sharedCapacity": 0,
  },
  "type": "@@MT/TICKETS/SET_TICKET_TOTAL_SHARED_CAPACITY",
}
`;

exports[`Gutenberg actions Set the status of the settings dashboard Close the settings dashboard 1`] = `
Object {
  "payload": Object {
    "isSettingsOpen": false,
  },
  "type": "@@MT/TICKETS/SET_TICKET_SETTINGS_OPEN",
}
`;

exports[`Gutenberg actions Set the status of the settings dashboard Close the settings dashboard 2`] = `
Object {
  "payload": Object {
    "isSettingsOpen": false,
  },
  "type": "@@MT/TICKETS/SET_TICKET_SETTINGS_OPEN",
}
`;

exports[`Gutenberg actions Set the status of the settings dashboard Open the settings dashboard 1`] = `
Object {
  "payload": Object {
    "isSettingsOpen": true,
  },
  "type": "@@MT/TICKETS/SET_TICKET_SETTINGS_OPEN",
}
`;

exports[`Gutenberg actions Set the status of the settings dashboard Open the settings dashboard 2`] = `
Object {
  "payload": Object {
    "isSettingsOpen": true,
  },
  "type": "@@MT/TICKETS/SET_TICKET_SETTINGS_OPEN",
}
`;

exports[`Gutenberg actions Single block actions Block registration process 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_TICKET_BLOCK_ID",
}
`;

exports[`Gutenberg actions Single block actions Block registration process 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/REMOVE_TICKET_BLOCK",
}
`;

exports[`Gutenberg actions Single block actions Create a new ticket 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_CREATE_NEW_TICKET",
}
`;

exports[`Gutenberg actions Single block actions Set ticket date is pristine 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "dateIsPristine": false,
  },
  "type": "@@MT/TICKETS/SET_TICKET_DATE_PRISTINE",
}
`;

exports[`Gutenberg actions Single block actions Set ticket date is pristine 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "dateIsPristine": true,
  },
  "type": "@@MT/TICKETS/SET_TICKET_DATE_PRISTINE",
}
`;

exports[`Gutenberg actions Single block actions Set ticket post ID 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "ticketId": 99,
  },
  "type": "@@MT/TICKETS/SET_TICKET_ID",
}
`;

exports[`Gutenberg actions Single block actions Ticket Capacity 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "capacity": "unlimited",
  },
  "type": "@@MT/TICKETS/SET_TICKET_CAPACITY",
}
`;

exports[`Gutenberg actions Single block actions Ticket Capacity 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "capacityType": "unlimited",
  },
  "type": "@@MT/TICKETS/SET_TICKET_CAPACITY_TYPE",
}
`;

exports[`Gutenberg actions Single block actions Ticket SKU 1`] = `
Object {
  "payload": Object {
    "SKU": "my-sku",
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_TICKET_SKU",
}
`;

exports[`Gutenberg actions Single block actions Ticket dates 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "startDate": "January 1, 2018",
  },
  "type": "@@MT/TICKETS/SET_TICKET_START_DATE",
}
`;

exports[`Gutenberg actions Single block actions Ticket dates 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "startTime": "10:00",
  },
  "type": "@@MT/TICKETS/SET_TICKET_START_TIME",
}
`;

exports[`Gutenberg actions Single block actions Ticket dates 3`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "endDate": "January 10, 2018",
  },
  "type": "@@MT/TICKETS/SET_TICKET_END_DATE",
}
`;

exports[`Gutenberg actions Single block actions Ticket dates 4`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "endTime": "12:34",
  },
  "type": "@@MT/TICKETS/SET_TICKET_END_TIME",
}
`;

exports[`Gutenberg actions Single block actions Ticket description 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "description": "The Next Generation of Digital Agency",
  },
  "type": "@@MT/TICKETS/SET_TICKET_DESCRIPTION",
}
`;

exports[`Gutenberg actions Single block actions Ticket editing flag 1`] = `
Object {
  "payload": Object {
    "blockId": true,
    "isEditing": undefined,
  },
  "type": "@@MT/TICKETS/SET_TICKET_IS_EDITING",
}
`;

exports[`Gutenberg actions Single block actions Ticket editing flag 2`] = `
Object {
  "payload": Object {
    "blockId": false,
    "isEditing": undefined,
  },
  "type": "@@MT/TICKETS/SET_TICKET_IS_EDITING",
}
`;

exports[`Gutenberg actions Single block actions Ticket end moment object 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "endDateMoment": Object {
      "type": "moment",
    },
  },
  "type": "@@MT/TICKETS/SET_TICKET_END_DATE_MOMENT",
}
`;

exports[`Gutenberg actions Single block actions Ticket end moment object 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "endDateMoment": null,
  },
  "type": "@@MT/TICKETS/SET_TICKET_END_DATE_MOMENT",
}
`;

exports[`Gutenberg actions Single block actions Ticket has been created 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "hasBeenCreated": true,
  },
  "type": "@@MT/TICKETS/SET_TICKET_HAS_BEEN_CREATED",
}
`;

exports[`Gutenberg actions Single block actions Ticket has been created 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "hasBeenCreated": false,
  },
  "type": "@@MT/TICKETS/SET_TICKET_HAS_BEEN_CREATED",
}
`;

exports[`Gutenberg actions Single block actions Ticket is loading 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "isLoading": true,
  },
  "type": "@@MT/TICKETS/SET_TICKET_IS_LOADING",
}
`;

exports[`Gutenberg actions Single block actions Ticket is loading 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "isLoading": false,
  },
  "type": "@@MT/TICKETS/SET_TICKET_IS_LOADING",
}
`;

exports[`Gutenberg actions Single block actions Ticket price 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "price": 99,
  },
  "type": "@@MT/TICKETS/SET_TICKET_PRICE",
}
`;

exports[`Gutenberg actions Single block actions Ticket start moment object 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "startDateMoment": Object {
      "type": "moment",
    },
  },
  "type": "@@MT/TICKETS/SET_TICKET_START_DATE_MOMENT",
}
`;

exports[`Gutenberg actions Single block actions Ticket start moment object 2`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "startDateMoment": null,
  },
  "type": "@@MT/TICKETS/SET_TICKET_START_DATE_MOMENT",
}
`;

exports[`Gutenberg actions Single block actions Update ticket 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/SET_UPDATE_TICKET",
}
`;

exports[`Gutenberg actions Single block actions cancel ticket edit 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
  },
  "type": "@@MT/TICKETS/CANCEL_EDIT_OF_TICKET",
}
`;

exports[`Gutenberg actions Single block actions fetch ticket details 1`] = `
Object {
  "payload": Object {
    "blockId": "modern-tribe",
    "ticketId": 99,
  },
  "type": "@@MT/TICKETS/FETCH_TICKET_DETAILS",
}
`;

exports[`Gutenberg actions Temporarily values Temporarily capacity 1`] = `
Object {
  "payload": Object {
    "sharedCapacity": 99,
  },
  "type": "@@MT/TICKETS/SET_TICKET_TMP_TICKET_SHARED_CAPACITY",
}
`;

exports[`Gutenberg actions Ticket Provider Set default provider 1`] = `
Object {
  "payload": Object {
    "provider": "Tribe__Tickets__Commerce__PayPal__Main",
  },
  "type": "@@MT/TICKETS/SET_PROVIDER",
}
`;
